### DISCUSSION TITLE
Обнуление плейлиста

### DISCUSSION THREAD

**Discussion by Bondian** [Chimildic/goofy#209] [Помощь с ошибкой]
Неожиданно исчезли все треки из плейлиста. Треки добавляются вручную. По триггеру лишь удалялись надоевшие треки, отправленные в бан. По данному тригеру в логах ошибок нет.

```
  let playlistWhite = Source.getPlaylistTracks('(((⚪))) БЕЛЫЙ', '3z2PhrpI1nyJCfSz0daSlY');
  Filter.removeTracks(playlistWhite, allBannedTracks);

  Playlist.saveWithUpdate({
    id: '3z2PhrpI1nyJCfSz0daSlY',
    name: '(((⚪))) БЕЛЫЙ ( ' + playlistWhite.length + ' )',
    public: true,
    tracks: playlistWhite,
    description: Playlist.getDescription(playlistWhite),
  });
```
Был раньше подобный случай с таким же плейлистом, но я подумал, что это произошло из-за перебора суточного лимита количества запросов и в связи с этим сбоя работы Playlist.saveWithUpdate (задавал ранее вопрос по работе Playlist.saveWithUpdate зесь: https://github.com/Chimildic/goofy/discussions/199). Но я принял меры и количество запросов давно не превышает лимит.
Я понимаю, что это вопрос практически без вводных данных, который невозможно воспроизвести. Но пишу в надежде на то, что кто-то уже раньше сталкивался с подобным, и есть вероятность обнаружить общие закономерности...

Есть ещё масса подобных плейлистов, но там всё в порядке...

**Comment by Chimildic** [comment]
Обычно из-за `saveWithUpdate` приходили раза два с сообщением что [терялась сортировка](https://github.com/Chimildic/goofy/discussions/119). Про частичную или полную потерю треков речи не шло никогда, насколько я помню.  В любом случае, 99,9% запусков не дают поймать подобные ошибки. И логических ошибок в этой функции я так и не нашел.   Поэтому виним Spotify, пока не можем доказать обратное.

Исходя из вашего кода, не понятно как собирается `allBannedTracks`. Поэтому кажется, что в этот массив каким-то образом попали треки из `playlistWhite`. Отсюда фильтр их и вычистил. 

Попробуйте в поддержку Spotify написать, возможно они умеют предыдущие ревизии плейлистов восстанавливать.

**Reply by Bondian** [reply]
В allBannedTracks выпавших треков нет.
Потерянные треки отловились в плейлисте, который собирал лайки, не разнесённые по цветам (я делю всю свою библиотеку по звучанию на условные цветные плейлисты. Обнулившийся плейлист был одним из таких "цветных" плейлистов). Там же оказались треки, пропажу которых я не заметил в других "цветных" плейлистах. 
Появились подозрения, что выпадение треков произошло после добавления в плейлисты локальных треков из телефона.. Наблюдаю дальше...
